quantus/metrics/base_batched.py

```diff
@@ -0,0 +1,404 @@
+        device (string): Indicated the device on which a torch.Tensor is or will be allocated: "cpu" or "gpu".
+        softmax (boolean): Indicates wheter to use softmax probabilities or logits in model prediction.
+            This is used for this __call__ only and won't be saved as attribute. If None, self.softmax is used.
+        model_predict_kwargs (dict, optional): Keyword arguments to be passed to the model's predict method.
+        batch_size (int): batch size for evaluation, default = 64.
+        custom_batch: any
+            Any object that can be passed to the evaluation process.
+            Gives flexibility to the user to adapt for implementing their own metric.
+
+        Returns
+        -------
+        last_results: a list of float(s) with the evaluation outcome of concerned batch
+
+        Examples
+        --------
+        # Enable GPU.
+        >> device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+
+        # Load a pre-trained LeNet classification model (architecture at quantus/helpers/models).
+        >> model = LeNet()
+        >> model.load_state_dict(torch.load("tutorials/assets/mnist"))
+
+        # Load MNIST datasets and make loaders.
+        >> test_set = torchvision.datasets.MNIST(root='./sample_data', download=True)
+        >> test_loader = torch.utils.data.DataLoader(test_set, batch_size=24)
+
+        # Load a batch of inputs and outputs to use for XAI evaluation.
+        >> x_batch, y_batch = iter(test_loader).next()
+        >> x_batch, y_batch = x_batch.cpu().numpy(), y_batch.cpu().numpy()
+
+        # Generate Saliency attributions of the test set batch of the test set.
+        >> a_batch_saliency = Saliency(model).attribute(inputs=x_batch, target=y_batch, abs=True).sum(axis=1)
+        >> a_batch_saliency = a_batch_saliency.cpu().numpy()
+
+        # Initialise the metric and evaluate explanations by calling the metric instance.
+        >> metric = Metric(abs=True, normalise=False)
+        >> scores = metric(model=model, x_batch=x_batch, y_batch=y_batch, a_batch=a_batch_saliency}
+        """
+        # Run deprecation warnings.
+        warn_func.deprecation_warnings(kwargs)
+        warn_func.check_kwargs(kwargs)
+
+        data = self.general_preprocess(
+            model=model,
+            x_batch=x_batch,
+            y_batch=y_batch,
+            a_batch=a_batch,
+            s_batch=s_batch,
+            custom_batch=custom_batch,
+            channel_first=channel_first,
+            explain_func=explain_func,
+            explain_func_kwargs=explain_func_kwargs,
+            model_predict_kwargs=model_predict_kwargs,
+            softmax=softmax,
+            device=device,
+        )
+
+        # create generator for generating batches
+        batch_generator = self.generate_batches(
+            data=data, batch_size=batch_size,
+            display_progressbar=self.display_progressbar,
+        )
+
+        # TODO: initialize correct length of last results
```

Comment:
Are these completed or still left to do? Is it necessary to fix this to merge/ production

Reply:
no not necessary

Reply:
removed the TODO